topdown/crypto.go

```diff
@@ -56,7 +56,23 @@ func builtinCryptoX509ParseCertificates(_ BuiltinContext, operands []*ast.Term,
 		return err
 	}
 
-	v, err := ast.InterfaceToValue(certs)
+	// add a field to certs containing the URIs as strings
+	processedCerts := make([]struct {
+		x509.Certificate
+		RawURIs []string `json:"RawURIs"`
```

Comment:
One comment about naming: I think the "raw" in `RawURIs` might give the impression that these are presented exactly as they appear in the subjectAltName extension. However, in general, passing a string through [`url.Parse`](https://pkg.go.dev/net/url#Parse) and then [`URL.String`](https://pkg.go.dev/net/url#URL.String) will not always result in the original string.[^1]

What do you think about naming this field `URIStrings` instead? (Or something else that doesn't have "raw" in the name?)

[^1]: Here are a few pathological examples: https://go.dev/play/p/I3cnmN5W6dy. It's possible that this shouldn't ever matter if the subjectAltName extension conforms to all the requirements of RFC 5280 [4.2.1.6](https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6), but I don't really understand this well enough to say for sure.

Reply:
This is a good point, I've updated the field name in f71330c342ab1de9a5489caad365eac3eccf2e02